### PR TITLE
Add installOSS flag in AmbassadorInstallation.Spec

### PR DESCRIPTION
**Pull request by concaf**
Installs [Ambassador OSS](https://www.getambassador.io/docs/latest/topics/install/install-ambassador-oss/)
instead of [AES](https://www.getambassador.io/docs/latest/topics/install/).

Default is false which means it installs AES by default.

### CODE REVIEW

File: pkg/controller/ambassadorinstallation/reconcile.go [hunk 1]
```diff
@@ -215,6 +225,16 @@ func (r *ReconcileAmbassadorInstallation) Reconcile(request reconcile.Request) (
 		helmValues["image.tag"] = tag
 	}
 
+	if ambObj.Spec.InstallOSS {
+		helmValues["enableAES"] = "false"
+		// We do not want to update image.repository and image.tag if they have already been populated by user supplied
+		// configuration.
+		if len(helmValues["image.repository"]) == 0 && len(helmValues["image.tag"]) == 0 {
+			helmValues["image.repository"] = "quay.io/datawire/ambassador"
+			helmValues["image.tag"] = "1.4.2"
```
Comment (concaf) [inline pkg/controller/ambassadorinstallation/reconcile.go:234]: @inercia do you think I should set `image.tag` here? `image.repository` is the only thing that differs between AES and OSS. If the default `image.tag` is used from the Helm repository, then we don't have to update this field here every time there is a new release. WDYT?

Comment (inercia) [reply pkg/controller/ambassadorinstallation/reconcile.go:234]: Yes, you are right. I think we should only set the `repository`. That way, a Helm chart that matches the `version` in the `AmbassadorInstallation` will be downloaded, and the `image.tag` will be obtained from that chart.

File: pkg/controller/ambassadorinstallation/reconcile.go [hunk 2]
```diff
@@ -198,6 +198,16 @@ func (r *ReconcileAmbassadorInstallation) Reconcile(request reconcile.Request) (
...
 	}
 
+	// `enableAES: true` means `installOSS: false`
+	// `enableAES: false` means `installOSS: true`
+	// Throw an error if these fields are inconsistent with each other
+	if (ambObj.Spec.HelmValues["enableAES"] == "false" && ambObj.Spec.InstallOSS == false) ||
+		(ambObj.Spec.HelmValues["enableAES"] == "true" && ambObj.Spec.InstallOSS == true) {
+		log.Info("helmValues.enableAES and installOSS fields conflict with each other",
+			"enableAES", ambObj.Spec.HelmValues["enableAES"], "installOSS", ambObj.Spec.InstallOSS)
+		return reconcile.Result{}, fmt.Errorf("helmValues.enableAES and installOSS fields conflict with each other")
```
Comment (inercia) [inline pkg/controller/ambassadorinstallation/reconcile.go:208]:
```suggestion
              status.SetCondition(ambassador.AmbInsCondition{
                                  Type:    ambassador.ConditionReleaseFailed,
                                  Status:  ambassador.StatusTrue,
                                  Reason:  ambassador.ReasonParametersError,
                                  Message: fmt.Sprintf("helmValues.enableAES and installOSS fields conflict with each other"),
              })
              _ = r.updateResourceStatus(ambIns, status)
              return reconcile.Result{}, err
```

File: pkg/controller/ambassadorinstallation/reconcile.go [hunk 3]
```diff
@@ -198,6 +198,23 @@ func (r *ReconcileAmbassadorInstallation) Reconcile(request reconcile.Request) (
...
+		log.Info("helmValues.enableAES and installOSS fields conflict with each other",
+			"enableAES", ambObj.Spec.HelmValues["enableAES"], "installOSS", ambObj.Spec.InstallOSS)
+              status.SetCondition(ambassador.AmbInsCondition{
+                                  Type:    ambassador.ConditionReleaseFailed,
+                                  Status:  ambassador.StatusTrue,
+                                  Reason:  ambassador.ReasonParametersError,
+                                  Message: fmt.Sprintf("helmValues.enableAES and installOSS fields conflict with each other"),
+              })
+              _ = r.updateResourceStatus(ambIns, status)
+              return reconcile.Result{}, err
```
Comment (concaf) [inline pkg/controller/ambassadorinstallation/reconcile.go:215]:
```suggestion
              return reconcile.Result{}, fmt.Errorf("helmValues.enableAES and installOSS fields conflict with each other")
```

File: ci/common.sh [hunk 4]
```diff
@@ -769,6 +769,12 @@ amb_get_image_tag() {
 	echo "$(amb_get_image $@)" | rev | cut -d ":" -f1 | rev
 }
 
+# get the repository part of the image in the Ambassador deployment
+# (ie, "quay.io/datawire/aes:1.1.0" -> quay.io/datawire/aes)
+amb_get_image_repository() {
+	echo "$(amb_get_image $@)" | cut -d ":" -f1
```
Comment (inercia) [inline ci/common.sh:775]: I think this will be unreliable. `registry.local:5000/ambassador:latest` will return `registry.local`

Comment (inercia) [reply ci/common.sh:775]:
```suggestion
	echo "$(amb_get_image $@)" | rev | cut -d ":" -f2- | rev
```

**Pull request merged**